generic/README.md

```diff
@@ -0,0 +1,233 @@
+
+    void demoMethod(ExperimentAPI experiment){
+        // Some code 
+        // Do something
+        // Do other things
+    }
+}
+```
+This can be achieved by adding a rule in the `input_rules.toml` file (as shown below) :
+```
+[[rules]]
+name = "Enum Based, toggle enabled"
+query = """((
+	(method_invocation 
+    	name : (_) @n1
+        arguments: ((argument_list 
+        				([
+                          (field_access field: (_)@f)
+                          (_) @f
+                         ])) )
+        	
+    ) @mi
+)
+(#eq? @n1 "isTreated")
+(#eq? @f "@stale_flag_name")
+)"""
+replace_node = "mi"
+replace = "@treated"
+groups = [ "replace_expression_with_boolean_literal"]
+holes = ["treated", "stale_flag_name"]
+```
+This specifies a rule that matches against expressions like `exp.isTreated(SOME_FLAG_NAME)` and replaces it with `true` or `false`. 
+The `query` property of the rule contains a [tree-sitter query](https://tree-sitter.github.io/tree-sitter/using-parsers#pattern-matching-with-queries) that is matched against the source code. 
+The node captured by the tag-name specified in the `replace_node` property is replaced with the pattern specified in the `replace` property.
+The `replace` pattern can use the tags from the `query` to construct a replacement based on the match (like [regex-replace](https://docs.microsoft.com/en-us/visualstudio/ide/using-regular-expressions-in-visual-studio?view=vs-2022)).
+
+The `rule` contains `holes` or template variables that need to be instantiated.
+For instance, in the above rule `@treated` and `@stale_flag_name` need to be replaced with some concrete value so that the rule matches only the feature flag API usages corresponding to a specific flag, and replace it specifically with `true` or `false`.  To specify such a behavior,
+user should create a `piranha_arguments.toml` file as shown below (assuming that the behavior of STALE_FLAG is **treated**): 
+```
+language = ["java"]
+substitutions = [
+    ["stale_flag_name", "STALE_FLAG"],
+    ["treated", "true"]
+]
+```
+This file specifies that, the user wants to perform this refactoring for `java` files. 
+The `substitutions` field captures mapping between the tags and their corresponding concrete values. In this example, we specify that the tag named `stale_flag_name` should be replaced with `STALE_FLAG` and `treated` with `true`.
+
+
+
+Each rule also contains the `groups` property, that specifies the kind of change performed by this rule. Based on this group, appropriate 
+cleanup rules will be performed by Piranha. For instance, `replace_expression_with_boolean_literal` will trigger deep cleanups (like eliminating `consequent` of a `if statement`) to eliminate dead code caused by replacing an expression with a boolean literal. 
+Currently, Piranha provides deep clean-ups for edits that belong the groups - `replace_expression_with_boolean_literal`, `delete_statement`, and `delete_method`. 
+
+## Onboarding a new language 
```

Comment:
Another section for Onboarding a new feature flag system?

Reply:
What do u mean?